src/pages/Governance/hooks/useSubmitVote.ts

```diff
@@ -0,0 +1,64 @@
+import {Types} from "aptos";
+import {useState} from "react";
+
+export type VoteResponse = VoteResponseOnSuccess | VoteResponseOnFailure;
+
+export type VoteResponseOnSuccess = {
+  succeeded: boolean;
+  transactionHash: string;
+};
+
+export type VoteResponseOnFailure = {
+  succeeded: boolean;
+  message: string;
+};
+
+const processTransaction = async (
+  transactionPayload: Types.TransactionPayload,
+): Promise<VoteResponse> => {
+  const responseOnFailure = {
+    succeeded: false,
+    message: "Unknown Error",
+  };
+
+  try {
+    const response = await window.aptos.signAndSubmitTransaction(
+      transactionPayload,
+    );
+    if ("hash" in response) {
+      return {
+        succeeded: true,
+        transactionHash: response["hash"],
+      };
+    }
+  } catch (error: any) {
+    if (typeof error == "object" && "message" in error) {
+      responseOnFailure.message = error.message;
+    }
+  }
+  return responseOnFailure;
+};
+
+// TODO: confirmation modal
+const useSubmitVote = (proposalId: string) => {
+  const [voteResponse, setVoteResponse] = useState<VoteResponse | null>(null);
+
+  async function submitVote(shouldPass: boolean, ownerAccountAddr: string) {
```

Comment:
not a blocker can revisit - I believe we are going toward BCS transaction
https://github.com/aptos-labs/wallet/pull/85#discussion_r938440809